Demo/Demo/FeatureViewControllers/FeatureBaseViewController.swift

```diff
@@ -87,49 +97,18 @@ class FeatureBaseViewController: UIViewController {
         DemoMerchantAPI.sharedService.createOrder(orderParams: orderRequestParams) { result in
             switch result {
             case .success(let order):
-                self.updateTitle("Order ID: \(order.id)")
-                self.orderID = order.id
+                self.baseViewModel.updateTitle("Order ID: \(order.id)")
+                DispatchQueue.main.async {
```

Comment:
why do we need to use `DispatchQueue.main.async` here?

Reply:
Without this since the orderID is now in a view model to be shared among our 2 views, we get the error `Publishing changes from background threads is not allowed; make sure to publish values from the main thread (via operators like receive(on:)) on model updates.`